### ISSUE TITLE
`pub run` should not emit any output not from the executable

### ISSUE THREAD

**Issue by jathak** [dart-lang/pub#2545]
# Environment

- `pub version` or `flutter pub version`: Pub 2.9.0-19.0-dev
- OS version: CI on Ubuntu, macOS, and Windows, locally on Debian
- Are you using the Chinese community mirror or a corporate firewall? No

# Problem

[Our tests are failing](https://travis-ci.org/github/google/dart_cli_pkg/builds/702222794) when run against the current dev version of the Dart SDK because `pub run` has started emitting output related to precompilation and we're expecting only output from the executable we're trying to run.

This issue is not present in the current stable SDK.

## Expected behavior

`pub run <executable>` should only emit output from the executable itself.

## Actual behavior

`pub run <executable>` first emits `Precompiling executable...` and other related lines.

**Comment by nex3** [comment]
This is also breaking our ability to use the protobuf compiler; see https://travis-ci.com/github/sass/dart-sass-embedded/jobs/363118689.

**Comment by sigurdm** [comment]
I believe this is a duplicate of https://github.com/dart-lang/pub/issues/2525
It was fixed in https://github.com/dart-lang/pub/commit/a8e2442b that should be included in 2.10.0-0.0.dev .

**Comment by alextekartik** [comment]
Even with "only print if there is a console attached" scenario, if any callers relies on stdout it will break.

In travis for example, My all dart setup (that dart file was writing a .rc file and printing the tmp filename to source) was:

```yaml
before_script:
  - source $(dart pub run xxxx)
```

In travis for example, My all dart setup (that dart file was writing a .rc file and printing the tmp filename to source) was:
I had to add a warmup calls just to workaround this issue
```yaml
before_script:
  # Dummy initial call
  - dart pub run xxxx --help
  - source $(dart pub run xxxx)
```

As I change sdk often, just calling `stagehand` sometimes displays:
```
$ stagehand --version
stagehand version: 3.3.9
```
and sometimes

```
$ stagehand --version
Can't load Kernel binary: Invalid SDK hash.
Precompiling executable... 
Precompiled stagehand:stagehand.
stagehand version: 3.3.9
```

Why not printing to stderr instead?
